packages/react/src/components/Dialog/Dialog.tsx

```diff
@@ -285,7 +286,8 @@ class Dialog extends AutoControlledComponent<WithAsProp<DialogProps>, DialogStat
 
     const dialogContent = (
       <Ref innerRef={this.contentRef}>
-        <ElementType
+        <ProviderBox
```

Comment:
My proposal would be to give guidance on the consumers to use the `ProviderBox` on their side (that was our suggestion with the `PopupContent` component. If we decide to change this, we should address it consistently in all portal components.